graph/src/data/schema.rs

```diff
@@ -73,8 +74,201 @@ pub enum SchemaValidationError {
+pub enum FulltextLanguage {
+    Simple,
+    Danish,
+    Dutch,
+    English,
+    Finnish,
+    French,
+    German,
+    Hungarian,
+    Italian,
+    Norwegian,
+    Portugese,
+    Romanian,
+    Russian,
+    Spanish,
+    Swedish,
+    Turkish,
+}
+
+impl TryFrom<&String> for FulltextLanguage {
+    type Error = String;
+    fn try_from(language: &String) -> Result<Self, Self::Error> {
+        match &language[..] {
+            "simple" => Ok(FulltextLanguage::Simple),
+            "da" => Ok(FulltextLanguage::Danish),
+            "nl" => Ok(FulltextLanguage::Dutch),
+            "en" => Ok(FulltextLanguage::English),
+            "fi" => Ok(FulltextLanguage::Finnish),
+            "fr" => Ok(FulltextLanguage::French),
+            "de" => Ok(FulltextLanguage::German),
+            "hu" => Ok(FulltextLanguage::Hungarian),
+            "it" => Ok(FulltextLanguage::Italian),
+            "no" => Ok(FulltextLanguage::Norwegian),
+            "pt" => Ok(FulltextLanguage::Portugese),
+            "ro" => Ok(FulltextLanguage::Romanian),
+            "ru" => Ok(FulltextLanguage::Russian),
+            "es" => Ok(FulltextLanguage::Spanish),
+            "sv" => Ok(FulltextLanguage::Swedish),
+            "tr" => Ok(FulltextLanguage::Turkish),
+            invalid => Err(format!(
+                "Provided language for fulltext search is invalid: {}",
+                invalid
+            )),
+        }
+    }
+}
+
+impl FulltextLanguage {
+    pub fn as_sql(&self) -> &'static str {
+        match self {
+            Self::Simple => "simple",
+            Self::Danish => "danish",
+            Self::Dutch => "dutch",
+            Self::English => "english",
+            Self::Finnish => "finnish",
+            Self::French => "french",
+            Self::German => "german",
+            Self::Hungarian => "hungarian",
+            Self::Italian => "italian",
+            Self::Norwegian => "norwegian",
+            Self::Portugese => "portugese",
+            Self::Romanian => "romanian",
+            Self::Russian => "russian",
+            Self::Spanish => "spanish",
+            Self::Swedish => "swedish",
+            Self::Turkish => "turkish",
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum FulltextAlgorithm {
+    Rank,
+    ProximityRank,
 }
 
+impl TryFrom<&String> for FulltextAlgorithm {
+    type Error = String;
+    fn try_from(algorithm: &String) -> Result<Self, Self::Error> {
+        match &algorithm[..] {
+            "rank" => Ok(FulltextAlgorithm::Rank),
+            "proximity_rank" => Ok(FulltextAlgorithm::ProximityRank),
+            invalid => Err(format!(
+                "Provided algorithm for fulltext search is invalid: {}",
+                invalid,
+            )),
+        }
+    }
+}
+
+impl FulltextAlgorithm {
+    pub fn as_sql(&self) -> &'static str {
+        match self {
+            Self::Rank => "ts_rank(",
+            Self::ProximityRank => "ts_rank_cd(",
```

Comment:
This only gets used once when building the order_by clause of a filter query, so it's a simple change to move the logic there. 
Though I like keeping `FulltextAlgorithm` manipulations contained to the trait, in this case it is _very_ Postgres specific.

Reply:
Ok, I've moved creating Postgres statements for FulltextAlgorithm into relational_queries within the `order_by()`.  
Similarly, there is a function for processing the language config into a string for use in sql; I've renamed that: `FulltextLanguage::as_sql()` -> `FulltextLanguage::as_str()` and kept it in graph::data::schema because it is generally applicable.